lib/iron_mq/message.rb

```diff
@@ -0,0 +1,92 @@
+require 'iron_mq/entity'
+
+module IronMQ
+  class Message
+    include Entity
+
+    define_fields :id, :body, :reservation_id, :reserved_count,
+                  :subscribers, :headers, :push_statuses
+
+    def initialize(client, queue_name, msg_fields)
+      @client = client
+      if queue_name.nil? || queue_name.empty?
+        fail ArgumentError, 'queue_name must be non-empty String'
+      end
+      @queue_name = queue_name
+      self.entity = msg_fields
+
+      fail ArgumentError, 'msg_fields must contain message ID' unless id
+    end
+
+    def get!
```

Comment:
This method mutates an object. That means it is dangerous.

```ruby
arr = [3, 7, 5, 9, 11]
result = arr.sort # => [3, 5, 7, 9, 11]
arr == result # => false
result = arr.sort! # => [3, 5, 7, 9, 11]
arr == result # => true
```

According to the library:

```ruby
require 'iron_mq'

client = IronMQ::Client.new
queue = client.queue('test-q')
msg = queue.post_message('my message', instantiate: true)
puts "Message: #{msg}"
api_response = msg.get! # it calls `self.entity = ...`, which mutates `msg` object
puts "Message: #{msg}"
puts "API response: #{api_response}"
```

In common, all methods of `IronMQ::Queue` and `IronMQ::Message`, which calls `self.entity = hash`, are mutates object. So that, all of them are "dangerous".